packages/eas-cli/src/submissions/ios/AppProduce.ts

```diff
@@ -0,0 +1,142 @@
+import { getConfig } from '@expo/config';
+import chalk from 'chalk';
+import wordwrap from 'wordwrap';
+
+import { authenticateAsync } from '../../credentials/ios/appstore/authenticate';
+import log from '../../log';
+import { promptAsync } from '../../prompts';
+import { IosSubmissionContext } from '../types';
+import { runFastlaneAsync, travelingFastlane } from '../utils/fastlane';
+import { validateLanguage } from './utils/language';
+
+/////////////////////////////////////////////////////////
+
+interface ProduceOptions {
+  appleId?: string;
+  appName?: string;
+  bundleIdentifier?: string;
+  appleTeamId?: string;
+  itcTeamId?: string;
+  language?: string;
+  companyName?: string;
+  sku?: string;
+}
+
+interface ProduceCredentials {
+  appleId?: string;
+  appleIdPassword?: string;
+  appleTeamId?: string;
+  itcTeamId?: string;
+  companyName?: string;
+  sku?: string;
+}
+
+type AppStoreResult = {
+  appleId: string;
+  appAppleId: string;
+};
+
+export async function ensureAppExistsAsync(ctx: IosSubmissionContext): Promise<AppStoreResult> {
```

Comment:
there is already `ensureAppExistsAsync` in `appstoreapi` directory, maybe `ensureAppStoreConnectAppExistsAsync` or sth else that suggests clearly what this function will do